components/system_notice/system_notice.jsx

```diff
@@ -16,53 +16,56 @@ export default class SystemNotice extends React.PureComponent {
         preferences: PropTypes.object.isRequired,
         dismissedNotices: PropTypes.object.isRequired,
         isSystemAdmin: PropTypes.bool,
+        serverVersion: PropTypes.string.isRequired,
+        config: PropTypes.object.isRequired,
+        license: PropTypes.object.isRequired,
+        analytics: PropTypes.object,
         actions: PropTypes.shape({
             savePreferences: PropTypes.func.isRequired,
             dismissNotice: PropTypes.func.isRequired,
+            getStandardAnalytics: PropTypes.func.isRequired,
         }).isRequired,
     }
 
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            currentNotice: null,
-        };
-    }
-
     componentDidMount() {
-        this.setCurrentNotice();
+        if (this.props.isSystemAdmin) {
+            this.props.actions.getStandardAnalytics();
+        }
     }
 
-    UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
-        this.setCurrentNotice(nextProps);
+    componentDidUpdate(prevProps) {
```

Comment:
nit: How likely is it that someone becomes a system admin without logging out and back in again?

Reply:
It doesn't happens often, but can happen if somebody promotes you to system admin while you are logged in (I think).